.github/workflows/update-website.yml

```diff
@@ -0,0 +1,41 @@
+name: Update Website
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'docs/src/**'
+
+jobs:
+  build_and_deploy:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    container: asterinas/asterinas:0.4.0
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: 'asterinas/asterinas'
+          path: 'asterinas'
+      
+      - name: Build the website
+        run: |
+          cd asterinas/docs
+          mdbook build
+
+      - name: Deploy website
+        env:
+          DEPLOY_KEY: ${{ secrets.DEPLOY_KEY }}
```

Comment:
`BOOK_PUBLISH_KEY` looks a better name than `DEPLOY_KEY` as the former name is more specific than the latter one.